testing/ios/IosUnitTests/build_and_run_tests.sh

```diff
@@ -1,7 +1,13 @@
 #!/bin/sh
+
+FLUTTER_ENGINE=ios_debug_sim_unopt
+
+if [ $# -eq 1 ]; then
+  FLUTTER_ENGINE=$1
+fi
+
 pushd $PWD
 cd ../../../..
-./flutter/tools/gn --ios --simulator --unoptimized
-ninja -j 100 -C out/ios_debug_sim_unopt
+ninja -j 100 -C out/$FLUTTER_ENGINE ios_test_flutter
```

Comment:
If we're still building the engine here, is there a reason we're not running the `gn` step? 
Given that `$FLUTTER_ENGINE` is now a variable, the `gn` params probably should be too.

Reply:
Instead of it being something it does, it's a precondition now.  I could pass the args through.  There isn't really a need other than to help people.  I added a check in the script with a helpful message.

Reply:
<tangential>just use `autoninja` here, so it knows if it should use goma or not.

Reply:
done